connector/connect/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -280,9 +280,22 @@ message RenameColumns {
   // Required. The input relation.
   Relation input = 1;
 
-  // Required.
+  // Optional.
   //
   // The number of columns of the input relation must be equal to the length
   // of this field. If this is not true, an exception will be returned.
+  //
+  // This field does not co-exist with the field `rename_columns_map`.
```

Comment:
maybe use `oneof`?
can this message cover all other `withColumnXXX` APIs?

Reply:
I thinks it will face the same thing as https://github.com/apache/spark/pull/38578#discussion_r1017875279 if use oneof. In that case it would be better to just split to have two independent messages. 

Maybe ENUM will be better?

Reply:
Example:

message RenameColumns {

ENUM {
   toDF;
   RenameColumnByNames;
   RenameColumnByExpressions;
}
  Enum type;
  field1_used_by_toDF;
  field2_used_by_RenameColumnByNames;
  field3_used_by_RenameColumnByExpressions;
}

Then based on the enum value we just fetch the corresponding fields?

Reply:
To me, it seems to have two separate approaches:

1. use enum to decide which field to use (and document those well).
2. split into independent messages.

Reply:
I personally prefer independent messages if the field can not be reused.
also cc @cloud-fan @HyukjinKwon

Reply:
I pushed a commit to split the message. For other `withColumnsXXX` we can add new messages for those by current approach.